main.go

```diff
@@ -12,5 +12,8 @@ func main() {
 
 	sf := NewSunfish()
 	defer sf.Close()
-	http.ListenAndServe(":"+*port, sf.Router)
+	err := http.ListenAndServe(":"+*port, sf.Router)
+	if err != nil {
+		panic(err)
```

Comment:
this doesn't seem like an appropriate place to panic.

Just print the error and quit.

panic should not be the default error handling mechanism, the default should be returning the error (and quitting if in main). Panic should only be used if it indicates that the developer has made a mistake. There are reasons that the http package could return errors that aren't developer mistakes.